### PR TITLE
JAVA-4868 3.12.x backport

### PR DESCRIPTION
**Pull request by ispringer**
JAVA-4868

### CODE REVIEW

File: driver-core/src/main/com/mongodb/internal/connection/ZstdCompressor.java [hunk 1]
```diff
@@ -46,7 +48,7 @@ public void compress(final List<ByteBuf> source, final BsonOutput target) {
 
         try {
             byte[] out = new byte[(int) Zstd.compressBound(uncompressedSize)];
-            int compressedSize = (int) Zstd.compress(out, singleByteArraySource, Zstd.maxCompressionLevel());
+            int compressedSize = (int) Zstd.compress(out, singleByteArraySource, DEFAULT_COMPRESSION_LEVEL);
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/connection/ZstdCompressor.java:51]:
```suggestion
            int compressedSize = (int) Zstd.compress(out, singleByteArraySource, Zstd.defaultCompressionLevel());
```

**Review by jyemin — CHANGES_REQUESTED** [review]

File: driver-core/src/main/com/mongodb/internal/connection/ZstdCompressor.java (hunk 1 above)
Comment (ispringer) [reply driver-core/src/main/com/mongodb/internal/connection/ZstdCompressor.java:51]: The `defaultCompressionLevel` was not added until a later version of zstd-jni. I considered upgrading to the latest zstd-jni but thought you might want to minimize additional changes. Let me know if you're ok with upgrading. You could consider it a backport of https://jira.mongodb.org/browse/JAVA-4965  :-}

Comment (jyemin) [reply driver-core/src/main/com/mongodb/internal/connection/ZstdCompressor.java:51]: Let's not do that.  It could easily break people's applications

**Comment by jyemin** [comment]
Merging, but not promising yet to actually release a 3.12 patch.

**Comment by ispringer** [comment]
> Merging, but not promising yet to actually release a 3.12 patch.

🤞

**Pull request merged**